django/core/handlers/asgi.py

```diff
@@ -0,0 +1,332 @@
+        query_string = self.scope.get('query_string', '')
+        if isinstance(query_string, bytes):
+            query_string = query_string.decode()
+        self.META = {
+            'REQUEST_METHOD': self.method,
+            'QUERY_STRING': query_string,
+            'SCRIPT_NAME': self.script_name,
+            'PATH_INFO': self.path_info,
+            # WSGI-expecting code will need these for a while
+            'wsgi.multithread': True,
+            'wsgi.multiprocess': True,
+        }
+        if self.scope.get('client'):
+            self.META['REMOTE_ADDR'] = self.scope['client'][0]
+            self.META['REMOTE_HOST'] = self.META['REMOTE_ADDR']
+            self.META['REMOTE_PORT'] = self.scope['client'][1]
+        if self.scope.get('server'):
+            self.META['SERVER_NAME'] = self.scope['server'][0]
+            self.META['SERVER_PORT'] = str(self.scope['server'][1])
+        else:
+            self.META['SERVER_NAME'] = 'unknown'
+            self.META['SERVER_PORT'] = '0'
+        # Headers go into META
+        for name, value in self.scope.get('headers', []):
+            name = name.decode('latin1')
+            if name == 'content-length':
+                corrected_name = 'CONTENT_LENGTH'
+            elif name == 'content-type':
+                corrected_name = 'CONTENT_TYPE'
+            else:
+                corrected_name = 'HTTP_%s' % name.upper().replace('-', '_')
+            # HTTP/2 say only ASCII chars are allowed in headers, but decode
+            # latin1 just in case.
+            value = value.decode('latin1')
+            if corrected_name in self.META:
+                value = self.META[corrected_name] + ',' + value
+            self.META[corrected_name] = value
+        # Pull out request encoding if we find it
+        if 'CONTENT_TYPE' in self.META:
+            self.content_type, self.content_params = cgi.parse_header(
+                self.META['CONTENT_TYPE']
+            )
+            if 'charset' in self.content_params:
+                try:
+                    codecs.lookup(self.content_params['charset'])
+                except LookupError:
+                    pass
+                else:
+                    self.encoding = self.content_params['charset']
+        else:
+            self.content_type, self.content_params = '', {}
+        # Pull out content length info
+        if self.META.get('CONTENT_LENGTH', None):
+            try:
+                self._content_length = int(self.META['CONTENT_LENGTH'])
+            except (ValueError, TypeError):
+                pass
+        # Save body
+        self._body = body
+        assert isinstance(self._body, bytes), 'Body is not bytes'
+        # Add a stream-a-like for the body
+        self._stream = BytesIO(self._body)
+        # Other bits
+        self.resolver_match = None
+
+    @cached_property
+    def GET(self):
+        return http.QueryDict(self.scope.get('query_string'))
```

Comment:
We process `query_string` in `__init__()`, so can we use instead `http.QueryDict(self.META['QUERY_STRING'])`?

Reply:
:+1: